### DISCUSSION TITLE
Gerbera 1.12.0 + Samsung 55" - Video always start from beginning

### DISCUSSION THREAD

**Discussion by regul8or** [gerbera/gerbera#2767] [Q&A] [answered]
Video always starts from the beginning. Table `grb_playstatus` have zeroes in corresponding columns (see attached screenshot).

<img width="336" alt="2022-12-25_185123" src="https://user-images.githubusercontent.com/21054394/209474921-2d77235d-ca31-4898-9ae9-5d41300741ae.png">

There are also other issues with this table. There's records for videos I didn't watch. And the following counts doesn't match:

```
SELECT count(*) FROM gerbera.grb_playstatus;
7
```

```
SELECT count(*)
FROM gerbera.grb_playstatus s, gerbera.mt_cds_object o
where s.item_id = o.id;
5
```

(I deleted all records from `grb_playstatus` for the experiment)

**Comment by KarlStraussberger** [comment, marked as answer]
Obviously, your tv is not recognized as Samsung. See either on clients page or in clients table. It's the userAgent value. You may have to set that in config.xml  (https://docs.gerbera.io/en/stable/config-clients.html)

**Reply by regul8or** [reply]
Looks like it **_is_** recognized

<img width="911" alt="2022-12-25_215257" src="https://user-images.githubusercontent.com/21054394/209479210-fc717503-1494-4550-80be-641c43d0e3c6.png">

**Reply by KarlStraussberger** [reply]
Good, but we do not know the exact way of bookmark handling for a specific series so we cannot set it by default, yet. Either set
`QUIRK_FLAG_SAMSUNG | QUIRK_FLAG_SAMSUNG_FEATURES | QUIRK_FLAG_SAMSUNG_HIDE_DYNAMIC | QUIRK_FLAG_SAMSUNG_BOOKMARK_SEC` or   `QUIRK_FLAG_SAMSUNG | QUIRK_FLAG_SAMSUNG_FEATURES | QUIRK_FLAG_SAMSUNG_HIDE_DYNAMIC | QUIRK_FLAG_SAMSUNG_BOOKMARK_MSEC` and report back that we can update our supported device list.

**Reply by regul8or** [reply]
Is it correct to put `<clients>` under `<config>` ?

```
<config>
  ...
  <clients>
    <client ip="192.168.1.144" flags="QUIRK_FLAG_SAMSUNG | QUIRK_FLAG_SAMSUNG_FEATURES | QUIRK_FLAG_SAMSUNG_HIDE_DYNAMIC | QUIRK_FLAG_SAMSUNG_BOOKMARK_MSEC"></client>
  </clients>
</config>
```

I edited `config.xml`, restarted Gerbera, opened it with TV. Web interface is still like a previous screenshot and it don't resume.

**Reply by KarlStraussberger** [reply]
That looks correct.

Does it show the new flags in web ui?

Can you search the debug output for `X_SetBookmark`?

**Reply by regul8or** [reply]
No, flags are not changed in Web UI

> Can you search the debug output for `X_SetBookmark`?

Where do I find it?

**Reply by KarlStraussberger** [reply]
You can start gerbera with --debug to get loads of messages including config checks and requests

**Reply by KarlStraussberger** [reply]
To make sure clients are active add `enabled="yes"`

**Reply by regul8or** [reply]
> To make sure clients are active add `enabled="yes"`

The result of adding `enabled="yes"` is very unexpected for me

<img width="913" alt="2022-12-27_195327" src="https://user-images.githubusercontent.com/21054394/209697521-96853069-4b5b-453a-a5e6-ed05b67dcc37.png">

Should I try `QUIRK_FLAG_SAMSUNG_BOOKMARK_SEC` instead of `...MSEC` or go straight to --debug?

**Reply by KarlStraussberger** [reply]
Sorry my fault, it has to be `SAMSUNG | SAMSUNG_FEATURES | SAMSUNG_HIDE_DYNAMIC | SAMSUNG_BOOKMARK_MSEC` in the config file.

**Reply by regul8or** [reply]
That's finally do the trick. Thank you

<img width="1742" alt="2022-12-27_221230" src="https://user-images.githubusercontent.com/21054394/209712313-b2c3ab98-5943-41f0-9a2c-e625416af82d.png">

In fact, it's not restarts from the precise moment it was stopped, it restarts a few moments before. Probably one day I will set `SAMSUNG_BOOKMARK_SEC` and it will be enough

**Reply by KarlStraussberger** [reply]
We're reducing the position by 10 seconds. Maybe should be configurable.

If MSEC works it's because of your TV sends bookmarks in milli-seconds. SEC will not work then.

**Comment by KarlStraussberger** [comment]
> There are also other issues with this table. There's records for videos I didn't watch. And the following counts doesn't match:
> 
> ```
> SELECT count(*) FROM gerbera.grb_playstatus;
> 7
> ```
> 
> ```
> SELECT count(*)
> FROM gerbera.grb_playstatus s, gerbera.mt_cds_object o
> where s.item_id = o.id;
> 5
> ```
> 
This may be caused by some deleted entries. Worth a look at the code.
